content/docs/for-buildpack-authors/concepts/build-plan.md

```diff
@@ -3,8 +3,43 @@ title="What is the build plan?"
 weight=99
 +++
 
+A **build plan** is a `toml` file is the output of the [detect](https://buildpacks.io/docs/for-buildpack-authors/concepts/lifecycle-phases/#phase-2-detect) phase, in which each image extension or component buildpack may express the dependencies it requires and the dependencies it provides.
+
 <!--more-->
 
-This page is a stub! The CNB project is applying to [Google Season of Docs](https://developers.google.com/season-of-docs/docs/timeline) to receive support for improving our documentation. Please check back soon.
+## Example Build Plan (TOML)
+
+In order to make contributions to the `Build Plan`, a `/bin/detect` executable MUST write entries to `<plan>` in two sections: `requires` and `provides`. The generated `plan.toml` file is usually added under the `<layers>`directory.
```

Comment:
We might clarify here, there are actually 3 different things that are relevant to the concept of a build plan:

* the piece of the build plan that is contributed by the buildpack during detect (this is the "build plan" from the buildpack's perspective)
  * this is written to a temporary directory by the buildpack
* the concatenation of all the contributions from all the buildpacks that passed detect (this is the real and true "build plan" from the platform's perspective)
  * this is the file that is usually written to `<layers>` (unless the platform provided another path for it) and the buildpack never has to actually care about it
* the piece of the build plan that is shown to the buildpack during build, containing only the dependencies that the buildpack is responsible for providing - and the buildpack can choose NOT to provide all of these, leaving that work for a future buildpack (this is sometimes referred to as the "buildpack plan")
  * this is written to a temporary directory by the lifecycle

Reply:
Right...from the buildpack-author perspective there's the stuff I contribute to the build plan during `detect` and then the _weird_ deduplication of the ["buildpackplan"](https://github.com/buildpacks/spec/blob/7207c7dd711e28ddd045e0130e8a96e9200bf079/buildpack.md#buildpack-plan-toml) that I have to perform during the `build` phase.

Reply:
@natalieparellano we also need to discuss this in more details

Reply:
https://buildpacks.io/docs/for-buildpack-authors/how-to/write-buildpacks/use-build-plan/ could become two pages: "write the build plan" and "read the buildpack plan"

Reply:
added